providers/terraform-provider-csbsqlserver/connector/connector.go

```diff
@@ -22,14 +22,23 @@ func New(server string, port int, username, password, database, encrypt string)
 }
 
 type Connector struct {
-	server, username, password, database, encrypt string
-	port                                          int
+	server   string
+	username string
+	password string
+	database string
+	encrypt  string
+	port     int
+}
+
+type databaseActor interface {
```

Comment:
**question**: Why do we have to introduce the `databaseActor`?

Reply:
The thinking is that although `sql.DB` and `sql.Tx` are very similar, they are different types. So a method such as `checkUser()` had to decide whether it would take a `sql.DB` or a `sql.Tx`, which seemed limiting as both implement the methods that it needs. This applies to the other helper methods, but in practice `checkUser()` is the only one that we call outside of a transaction, but another approach would have been to move that check into a transaction too.

Reply:
Thanks!

Reply:
It looks, though, that we're never passing the `sql.Tx` where we used to pass `sql.DB`.